icechunk/src/config.rs

```diff
@@ -71,12 +69,19 @@ impl RepositoryConfig {
 }
 
 #[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
-pub struct StaticCredentials {
+pub struct S3Credentials {
     pub access_key_id: String,
     pub secret_access_key: String,
     pub session_token: Option<String>,
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+pub enum GcsCredentials {
+    ServiceAccount(PathBuf),
+    ServiceAccountKey(String),
+    ApplicationCredentials(PathBuf),
+}
```

Comment:
Do we need `ServiceAccount`? Can they just pass the key string instead? Are there defaults for these paths as in AWS? Are some of these options replace by `FromEnv`? Maybe `FromEnv` needs extra optional options?

Reply:
Gcs world accepts json or serialized json, we need to support both